src/qibo/quantum_info/utils.py

```diff
@@ -1,18 +1,31 @@
 """Utility functions for the Quantum Information module."""
 
-from functools import reduce
+from functools import cache, reduce
 from itertools import permutations
 from math import factorial
 from re import finditer
 from typing import List, Optional, Tuple, Union
 
 import numpy as np
 
-from qibo import matrices
-from qibo.backends import _check_backend
+from qibo.backends import Backend, _check_backend
 from qibo.config import PRECISION_TOL, raise_error
 
 
+@cache
+def _get_paulis(order: str, backend: Backend):
```

Comment:
```suggestion
def _get_single_paulis(order: str, backend: Backend):
```

Or maybe `_get_paulis_single`. One or the other. Just to be very clear that it's single-qubit Paulis.